ansible/library/reduce_and_add_sonic_images.py

```diff
@@ -30,7 +30,9 @@
 
 def exec_command(module, cmd, ignore_error=False, msg="executing command"):
     rc, out, err = module.run_command(cmd)
-    if not ignore_error and rc != 0:
+    if ignore_error:
+        return rc, out, err
```

Comment:
You should restore the original code structure and change line 38 instead. With your change, the succeeded path will return different number of return values.